src/Components/Endpoints/src/Rendering/SSRRenderModeBoundary.cs

```diff
@@ -44,6 +50,50 @@ public SSRRenderModeBoundary([DynamicallyAccessedMembers(Component)] Type compon
         };
     }
 
+    private static void AssertRenderModeIsConfigured(HttpContext httpContext, Type componentType, IComponentRenderMode renderMode)
+    {
+        var configuredRenderModesMetadata = httpContext.GetEndpoint()?.Metadata.GetMetadata<ConfiguredRenderModesMetadata>();
+        if (configuredRenderModesMetadata is null)
+        {
+            // This is not a Razor Components endpoint. It might be that the app is using RazorComponentResult,
+            // or perhaps something else has changed the endpoint dynamically. In this case we don't know how
+            // the app is configured so we just proceed and allow any errors to happen if the client-side code
+            // later tries to reach endpoints that aren't mapped.
+            return;
```

Comment:
Semi-related: a possible way forwards for one of the `UseExceptionHandler` problems (https://github.com/dotnet/aspnetcore/issues/50287) might be here. If the error handling middleware leaves some clue that it's responsible for the current endpoint (e.g., if it drops some hint into `httpContext.Items`) then `SSRRenderModeBoundary` could simply behave as if there was no rendermode, thereby forcing the entire page to render SSR-only. The client-side code would no longer try to start a circuit/wasm-runtime because there would be no interactive component markers in the page.

That doesn't solve the "need a new DI scope" issue but it solves 50% of the problem at least.

Reply:
@SteveSandersonMS what if the page was able to say "Disallow interactivity" through an attribute or similar (only at the page level).

That would mean the Error page can apply that attribute and there's no dance needed between the exception handler and Blazor.

Reply:
Yes that's probably reasonable. Let's do any follow-up about that in https://github.com/dotnet/aspnetcore/issues/50287